### DISCUSSION TITLE
NDCG@K example code

### DISCUSSION THREAD

**Discussion by songsong0425** [pyg-team/pytorch_geometric#7792] [Ideas]
Dear PyG community,

Hi, always thank you for your effort in package maintenance and development.
When I tried to implement the link prediction metrics (i.e., hits@k, ndcg@k), I couldn't find the example code for ndcg@k.
Can you consider adding the example about the ndcg@k for the evaluation metrics?
If it has already existed and I missed it, can anyone tell me where can I find it?

Thank you for reading!

**Comment by rusty1s** [comment]
It's currently true that we don't have great support for any more sophisticated link prediction metrics other than `auprc` or `auroc`. Which example would you like to see implement these metrics?

**Reply by songsong0425** [reply]
Well... I didn't specify the example, if possible, https://github.com/pyg-team/pytorch_geometric/blob/master/examples/link_pred.py or https://github.com/pyg-team/pytorch_geometric/blob/master/examples/ppi.py will be the proper examples. Actually, there were some example codes for ndcg@k (e.g., https://github.com/zuoxingdong/recsys_metrics), but it would be more convenient if PyG has own implementation.

**Reply by rusty1s** [reply]
Got it. I agree, this would be super nice to have, although this is not necessarily easy to implement. Can we convert this to an issue and track progress there?

**Reply by songsong0425** [reply]
Sure. It will be great. Please feel free to update the related codes for it. :)

**Reply by rusty1s** [reply]
Can you take care of creating this issue for me? This would be highly appreciated.

**Reply by songsong0425** [reply]
Please forgive me for the late reply. I'm not familiar with Github, but is it correct that I need to create the issue by the `Reference in new issue` button?

**Reply by rusty1s** [reply]
Yes, that works. Alternatively, you can just create a new issue and link to here in its description.

**Reply by songsong0425** [reply]
As you suggested, I created the issue just now. Please check it and notify me if there is any problem. Thanks!
